File: src/test/java/cz/vhromada/catalog/web/common/SeasonUtils.java
package cz.vhromada.catalog.web.common;

import static org.assertj.core.api.SoftAssertions.assertSoftly;

import java.util.Collections;

import cz.vhromada.catalog.entity.Season;
import cz.vhromada.catalog.web.fo.SeasonFO;
import cz.vhromada.common.Language;

/**
 * A class represents utility class for seasons.
 *
 * @author Vladimir Hromada
 */
public final class SeasonUtils {

    /**
     * Creates a new instance of SeasonUtils.
     */
    private SeasonUtils() {
    }

    /**
     * Returns FO for season.
     *
     * @return FO for season
     */
    public static SeasonFO getSeasonFO() {
        final SeasonFO season = new SeasonFO();
        season.setId(CatalogUtils.ID);
        season.setNumber(Integer.toString(CatalogUtils.NUMBER));
        season.setStartYear(Integer.toString(CatalogUtils.YEAR));
        season.setEndYear(Integer.toString(CatalogUtils.YEAR + 1));
        season.setLanguage(Language.EN);
        season.setSubtitles(Collections.singletonList(Language.CZ));
        season.setNote(CatalogUtils.NOTE);
        season.setPosition(CatalogUtils.POSITION);

        return season;
    }

    /**
     * Returns season.
     *
     * @return season
     */
    public static Season getSeason() {
        final Season season = new Season();
        season.setId(CatalogUtils.ID);
        season.setNumber(CatalogUtils.NUMBER);
        season.setStartYear(CatalogUtils.YEAR);
        season.setEndYear(CatalogUtils.YEAR + 1);
        season.setLanguage(Language.EN);
        season.setSubtitles(Collections.singletonList(Language.CZ));
        season.setNote(CatalogUtils.NOTE);
        season.setPosition(CatalogUtils.POSITION);

        return season;
    }

    /**
     * Asserts season deep equals.
     *
     * @param expected expected FO for season
     * @param actual   actual season
     */
    public static void assertSeasonDeepEquals(final SeasonFO expected, final Season actual) {
        assertSoftly(softly -> {
            softly.assertThat(expected).isNotNull();
            softly.assertThat(actual).isNotNull();
        });
        assertSoftly(softly -> {
            softly.assertThat(actual.getId()).isEqualTo(expected.getId());
            softly.assertThat(actual.getNumber()).isEqualTo(Integer.parseInt(expected.getNumber()));
            softly.assertThat(actual.getStartYear()).isEqualTo(Integer.parseInt(expected.getStartYear()));
            softly.assertThat(actual.getEndYear()).isEqualTo(Integer.parseInt(expected.getEndYear()));
            softly.assertThat(actual.getLanguage()).isEqualTo(expected.getLanguage());
            softly.assertThat(actual.getSubtitles()).isEqualTo(expected.getSubtitles());
            softly.assertThat(actual.getNote()).isEqualTo(expected.getNote());
            softly.assertThat(actual.getPosition()).isEqualTo(expected.getPosition());
        });
    }

}

File: src/test/kotlin/cz/vhromada/catalog/web/common/SeasonUtils.kt
package cz.vhromada.catalog.web.common

import cz.vhromada.catalog.entity.Season
import cz.vhromada.catalog.web.fo.SeasonFO
import cz.vhromada.common.Language
import org.assertj.core.api.SoftAssertions.assertSoftly

/**
 * A class represents utility class for seasons.
 *
 * @author Vladimir Hromada
 */
object SeasonUtils {

    /**
     * Returns FO for season.
     *
     * @return FO for season
     */
    fun getSeasonFO(): SeasonFO {
        return SeasonFO(id = CatalogUtils.ID,
                number = CatalogUtils.NUMBER.toString(),
                startYear = CatalogUtils.YEAR.toString(),
                endYear = (CatalogUtils.YEAR + 1).toString(),
                language = Language.EN,
                subtitles = listOf(Language.CZ),
                note = CatalogUtils.NOTE,
                position = CatalogUtils.POSITION)
    }

    /**
     * Returns season.
     *
     * @return season
     */
    fun getSeason(): Season {
        return Season(id = CatalogUtils.ID,
                number = CatalogUtils.NUMBER,
                startYear = CatalogUtils.YEAR,
                endYear = CatalogUtils.YEAR + 1,
                language = Language.EN,
                subtitles = listOf(Language.CZ),
                note = CatalogUtils.NOTE,
                position = CatalogUtils.POSITION)
    }

    /**
     * Asserts season deep equals.
     *
     * @param expected expected FO for season
     * @param actual   actual season
     */
    fun assertSeasonDeepEquals(expected: SeasonFO?, actual: Season?) {
        assertSoftly {
            it.assertThat(expected).isNotNull
            it.assertThat(actual).isNotNull
        }
        assertSoftly {
            it.assertThat(actual!!.id).isEqualTo(expected!!.id)
            it.assertThat(actual.number).isEqualTo(Integer.parseInt(expected.number))
            it.assertThat(actual.startYear).isEqualTo(Integer.parseInt(expected.startYear))
            it.assertThat(actual.endYear).isEqualTo(Integer.parseInt(expected.endYear))
            it.assertThat<Language>(actual.language).isEqualTo(expected.language)
            it.assertThat<Language>(actual.subtitles).isEqualTo(expected.subtitles)
            it.assertThat(actual.note).isEqualTo(expected.note)
            it.assertThat(actual.position).isEqualTo(expected.position)
        }
    }

}
